sycl/test/extensions/properties/properties_kernel_device_has_warning.cpp

```diff
@@ -0,0 +1,167 @@
+// RUN: %clangxx -fsycl-device-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+
+// Tests for warnings when propagated aspects do not match the aspects available
+// in a function, as specified through the 'sycl::device_has' property.
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental;
+
+// expected-note-re@+1 4 {{propagated from call to function '{{.*}}Struct1::Struct1({{.*}})'}}
+struct [[__sycl_detail__::__uses_aspects__(aspect::fp16)]] Struct1 {
+  int a = 0;
+};
+
+// expected-note-re@+1 4 {{propagated from call to function '{{.*}}func5(int)'}}
```

Comment:
I don't think this particular check is meaningful enough if the originating callstack isn't verified together with it.

Reply:
I am not sure what you mean? It is checking that the diagnostic points to the line below and that it is issued 4 times, to make sure it happens the expected number of times. Do you mean that multiple of these could be from the same warning? If so you're right but I am not aware of a way to make sure they are happening in the right order while also making sure it happens at the right line, without making the test a headache to maintain.

Reply:
I wasn't aware of the "4" times syntax. I guess it's better than I thought, but just too scattered making reading it harder. No idea how to fix it though, so we'll live with that.